src/main/java/com/knu/buga1chuk/service/PersonGenerator.java

```diff
@@ -0,0 +1,80 @@
+package com.knu.buga1chuk.service;
+
+import com.knu.buga1chuk.serialization.Person;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+public class PersonGenerator {
+    private static final Random random = new Random();
+    private static int CURRENT_PERSON_ID = 0;
+
+    public static Person getRandomPerson() {
+
+        CURRENT_PERSON_ID++;
+        String name = generateRandomName();
+        int age = random.nextInt(88) + 12;
+        String city = generateRandomLocation();
+
+        return new Person(CURRENT_PERSON_ID, name, age, city);
+    }
+
+    public static List<String> getNames(String filePath) {
+        List<String> names = new ArrayList<>();
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(filePath))) {
+            String name;
+            while ((name = reader.readLine()) != null) {
+                names.add(name);
+            }
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+            System.out.println("There is no such file");
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        return names;
+    }
+
+    public static String generateRandomName() {
+
+        List<String> list = getNames("target/names");
+
+        int randomInt = random.nextInt(list.size());
+
+        return list.get(randomInt);
+    }
+
+    public static List<String> getCities(String filePath) {
+        List<String> cities = new ArrayList<>();
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(filePath))) {
+            String city;
+            while ((city = reader.readLine()) != null) {
+                cities.add(city);
+            }
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+            System.out.println("There is no such file");
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        return cities;
+    }
+
+    public static String generateRandomLocation() {
+        List<String> list = getCities("target/cities");
```

Comment:
We read the file every time we execute this method.
It will produce performance issues.

The names should be retrieved from the file only once, and then we can use it.

Reply:
Fix